### PR TITLE
feat: set any nvim_open_win option on hint window

### LINKED ISSUE #4: [feature request] window hint titles/footers
I want to allow users to pass `title`, `title_pos`, to the hint window.
Given that the options: `footer` and `footer_pos` are coming in version 0.10, adding the ability to specify arbitrary options that get passed to `:h nvim_open_win()` is probably the way to go as this makes things more future proof.

I'm thinking add an option to the `config` field called `float_opts`.

this would remove the need to explicitly handle border as a separate option, but we should still handle this option separately so we don't break people.

Would also need to add a new highlight group: `HydraTitle`. A highlight for the footer can be added once 10.0 is stable, or we can add it now behind a version check.

### PR DESCRIPTION
**Pull request by benlubas**
closes #4

Things to note:
- We might want to add a deprecation message to the `config.hint.border` option, as this should now be customized with `config.hint.float_opts.border`.
- I've opted for implementing the footer highlights, and I'm using `vim.version().minor >= 10` to set them.
- have tested on 9.4 and nightly

### CODE REVIEW

**Comment by miversen33** [comment]
I like the idea here but this does break existing configurations. Using your branch, I lose my border. I can of course change my config to utilize the new `float_opts` and it does of course work then. But I think we should be very careful about breaking stuff while moving forward. 

Neo-tree handles this by deprecating first and removing later, a tact I have taken on in my own projects as well. I think if we are going to move forward with the idea that `float_opts` is how we are going to configure the appearance of hydras, we must inform the user first before breaking their configuration. After all, breaking changes are bad

Hydra does not have any sort of deprecation system (that I am aware of), so we would probably want to add that.

Thoughts?

**Comment by benlubas** [comment]
Yup, definitely agree. I honestly just forgot to keep support for the old config, but had decided it was a good idea at the start.

**Comment by benlubas** [comment]
> Hydra does not have any sort of deprecation system (that I am aware of), so we would probably want to add that.

Adding versioning is probably the way to go, release-please is really nice I can set it up. 

As far as notifying users, we can just include a simple check and then use `vim.notify()` maybe add a global so it only triggers once (per nvim launch)

**Comment by miversen33** [comment]
I think versioning is a good idea as well. I have added a deprecation system to hydra but I cannot seem to push to your branch so I guess check out [my branch](https://github.com/nvimtools/hydra.nvim/compare/main...miversen33:hydra.nvim:add_deprecation_system) and see if we can merge that onto this. Alternatively, I can merge in the deprecation system and you can rebase onto main after its in place. IMO it should make handling stuff like this in the future better.

Also unrelated to that, we need to update the vimdoc as it still says "hint.border" and has no reference to "hint.float_opts".

[persisted.nvim](https://github.com/olimorris/persisted.nvim/blob/main/Makefile) uses a makefile to automagically update their vimdoc based on the readme, I wonder how difficult that would be to add here

**Comment by miversen33** [comment]
Deprecation system is merged, do you want to rebase this and use that?

**Comment by benlubas** [comment]
ah there's an issue, hint can be a boolean, so we need to type check it.

File: lua/hydra/lib/util.lua [hunk 1]
```diff
@@ -187,7 +188,8 @@ function util.warn_deprecated(option, config)
       if deprecation_notice.hint then
          message = string.format("%s -- See %s", message, deprecation_notice.hint)
       end
-      vim.notify(message, vim.log.levels.WARN, {
+      -- \n turns this into a "press-enter" message (:h press-enter), it's also the only way I can see the message
+      vim.notify(message .. "\n", vim.log.levels.WARN, {
```
Comment (benlubas) [inline lua/hydra/lib/util.lua:192]: I'm not quite sure why, but this message doesn't show for me on startup unless I turn it into a press enter message. I probably have some type of config messing with things. But this makes sure the message isn't missed.

**Comment by benlubas** [comment]
I'm going to look into the vim doc generation now

**Comment by benlubas** [comment]
I think it's best to just use the [github action](https://github.com/marketplace/actions/panvimdoc) instead of setting it up in a makefile.

**Comment by benlubas** [comment]
Reading through, that kinda clobbers a lot of good headers. I think that's just going to be temporary though, after the readme overhaul it should look a lot better

File: lua/hydra/lib/util.lua (hunk 1 above)
Comment (miversen33) [reply lua/hydra/lib/util.lua:192]: Thats odd though I actually kind of like the added friction. Deprecation warnings should be annoying.

**Comment by miversen33** [comment]
I am going to wait to further review this until all changes are done :) Feel free to ping me once you are good

**Comment by benlubas** [comment]
@miversen33 I think this is good to review. I'm going to do the readme stuff in a different PR I think, probably in a few parts

File: lua/hydra/lib/util.lua (hunk 1 above)
Comment (miversen33) [reply lua/hydra/lib/util.lua:192]: This is adding an extra newline to on vim.notify when using nvim-notify. I don't know that I care enough to try and "fix" it, but figured I would call it out

**Review by miversen33 — APPROVED** [review]
This all seems good. IMO its ready to merge

**Comment by miversen33** [comment]
I appreciate this, thanks for the contributions as well :)

**Pull request merged**